ui/app/components/totp-edit.js

```diff
@@ -18,6 +18,16 @@ export default class TotpEdit extends Component {
 
   @tracked hasGenerated = false;
   successCallback;
+  generatedDefaultFields = ['name', 'generate', 'issuer', 'accountName'];
+  nonGeneratedDefaultFields = [...this.generatedDefaultFields, 'url', 'key'];
+
+  get generatedFields() {
+    return this.generatedDefaultFields;
+  }
+
+  get nonGeneratedFields() {
+    return this.nonGeneratedDefaultFields;
+  }
```

Comment:
Nice! This is a lot easier to parse here instead of in the model 🤩 

Since the getters here are just returning static fields, we don't technically need them and could just reference `this.generatedDefaultFields` or `this.nonGeneratedDefaultFields` in the template. 

**However**, I think this is a good opportunity to leverage getters, but instead compute the return based on the value of `this.args.model.generate`:
```suggestion
  get keyFormFields() {
    const  generated = ['name', 'generate', 'issuer', 'accountName'];
    const  notGenerated = [...generated, 'url', 'key'];
    return this.args.model.generate ? generated : notGenerated;
  }
```